iphone/cli/commands/_build.js

```diff
@@ -5809,27 +5809,37 @@ iOSBuilder.prototype.copyResources = function copyResources(next) {
 									analyzeOptions.targets = { ios: this.minSupportedIosSdk }; // if using jscore, target our min ios version
 								} // if not jscore, just transpile everything down (no target)
 
-								const modified = jsanalyze.analyzeJs(source, analyzeOptions);
-								const newContents = modified.contents;
-
-								// we want to sort by the "to" filename so that we correctly handle file overwriting
-								this.tiSymbols[to] = modified.symbols;
-
-								const dir = path.dirname(to);
-								fs.existsSync(dir) || wrench.mkdirSyncRecursive(dir);
-
-								const exists = fs.existsSync(to);
-								// dest doesn't exist, or new contents differs from existing dest file
-								if (!exists || newContents !== fs.readFileSync(to).toString()) {
-									this.logger.debug(__('Copying and minifying %s => %s', from.cyan, to.cyan));
-									exists && fs.unlinkSync(to);
-									fs.writeFileSync(to, newContents);
-									this.jsFilesChanged = true;
-								} else {
-									this.logger.trace(__('No change, skipping %s', to.cyan));
+								try {
+									const modified = jsanalyze.analyzeJs(source, analyzeOptions);
+									const newContents = modified.contents;
+
+									// we want to sort by the "to" filename so that we correctly handle file overwriting
+									this.tiSymbols[to] = modified.symbols;
+
+									const dir = path.dirname(to);
+									fs.existsSync(dir) || wrench.mkdirSyncRecursive(dir);
+
+									const exists = fs.existsSync(to);
+									// dest doesn't exist, or new contents differs from existing dest file
+									if (!exists || newContents !== fs.readFileSync(to).toString()) {
+										this.logger.debug(__('Copying and minifying %s => %s', from.cyan, to.cyan));
+										exists && fs.unlinkSync(to);
+										fs.writeFileSync(to, newContents);
+										this.jsFilesChanged = true;
+									} else {
+										this.logger.trace(__('No change, skipping %s', to.cyan));
+									}
+									cb2();
+								} catch (err) {
+									err.message.split('\n').forEach(this.logger.error);
+									if (err.codeFrame) { // if we have a nicely formatted pointer to syntax error from babel, use it!
+										this.logger.log(err.codeFrame);
+									}
+									this.logger.log();
+									process.exit(1);
+								} finally {
+									this.unmarkBuildDirFile(to);
```

Comment:
Does it make a different to call `cb2()` before or after unmarking a file? It was called after it previously and is called after it now. Still in testing

Reply:
In order to surface the nicer error output, I catch the error and spit it out here and exit the process rather than propagating it up the callback stack.

the unmarking happens before the error handling.